vault/resource_config_ui_custom_messages_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"encoding/base64"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+)
+
+var (
+	messageBase64 = base64.StdEncoding.EncodeToString([]byte("Vault will be unavailable for maintenance on 2024-02-28 from 05:00Z to 07:00Z"))
+)
+
+func testConfigUICustomMessageConfig(isUpdate bool) string {
+	if !isUpdate {
+		return fmt.Sprintf(`
+		resource "vault_config_ui_custom_message" "test" {
+			title = "Maintenance Adviosry"
+			message_base64 = "%s"
+			start_time = "2024-02-01T00:00:00Z"						
+		}`, messageBase64) // There's an intentional typo in the title
+	} else {
+		return fmt.Sprintf(`
+		resource "vault_config_ui_custom_message" "test" {
+			title = "Maintenance Advisory"
+			message_base64 = "%s"
+			start_time = "2024-02-01T00:00:00Z"
+			end_time = "2024-02-27T23:59:59Z"
+			type = "modal"
+			authenticated = false
+			link {
+				title = "Learn more"
+				href = "https://www.hashicorp.com"
+			}
+			options = {
+				"background-color" = "red"
+			}
+		}`, messageBase64) // That intentional typo in the title is fixed here
+	}
+}
+
+func TestAccConfigUICustomMessage(t *testing.T) {
+	resourceName := "vault_config_ui_custom_message.test"
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: providerFactories,
+		PreCheck:          func() { testutil.TestAccPreCheck(t) },
```

Comment:
```suggestion
		PreCheck:          func() { testutil.TestAccPreCheck(t) },
		SkipIfAPIVersionLT(t, testProvider.Meta(), provider.VaultVersion116)

```

Reply:
This will make sure the test doesn't run against Vault versions that don't support this resource.